### ISSUE TITLE
Awesome!

### ISSUE THREAD

**Issue by jeff-h** [pascalpp/FreeRuler#1]
Super cool that you're re-doing this in Swift. I've used your REALbasic version for so many years I can't even remember :)

Let me know if you have any particular questions about reimplementing this in Cocoa and Swift. I'd love to try to help.

**Comment by pascalpp** [comment]
@jeff-h hey great! i started this this summer but had to put it aside shortly after WWDC because of day job stuff. Thanks for the PR— I haven't upgraded to XCode 7 yet, but will be doing so shortly.

**Comment by derwaldgeist** [comment]
Hi, any updates on this? :-)

**Comment by jeff-h** [comment]
With the imminent demise of 32-bit apps in macOS 10.15 I figured it was time to revisit this. I've created a basic working version over the weekend, albeit with just a horizontal ruler at present.

It's a completely fresh start, since nothing in the existing codebase compiles any more. However I did take inspiration from your code.

Are you still up for this project overall? If not I'll put it on my own Github account, but since you're the original author and I've been a fan of the original Free Ruler for so long, it seemed only right to put it here :)  PR is at https://github.com/pascalpp/FreeRuler/pull/3

Notes:
* only horizontal for now
* code is a bit disorganised. I just wanted to get this up and running, so the code could definitely be more beautiful, moving more code into enums, protocols, extensions etc
* no units other than pixels
* only keyboard shortcut implemented thus far are the arrow-keys

I thought the aesthetic, feature set and keyboard controls of the original were pretty much perfect, so would propose this be worked towards that standard. There are other things that could probably be done after that eg multiple rulers (rather than only ever 2) etc
